### ISSUE TITLE
Split monolithic elonacore.cpp into separate files

### ISSUE THREAD

**Issue by Ruin0x11** [ElonaFoobar/ElonaFoobar#284]
# Details
Before we can do many useful things with the code it should be split apart into modules of considerably smaller size. Trying to move things around and making other refactorings at the same time makes the resulting PR impossible to follow, so things should be moved out of `elonacore.cpp` only, and no other code changes should be made. Following that there should be separate PRs for each refactored separate file to make the process of bug detection far easier.

# Tasks

- Create new files splitting out the code in `elonacore.cpp` and create a new PR for every new file created.
- Move some functionality to existing files that make sense semantically.
- Make some variables global using `ELONA_EXTERN` in `variables.hpp` for the purpose of getting the code to compile. This will only be temporary as the variables will either be localized or moved into container structs encapsulating them in future changes.
- Do not make any changes to the code except moving globals/functions.
- At the very end, organize files in a nested directory structure. For example, grouping all databases together.
- Make sure the PRs are short enough to comprehend.
- As the PRs are merged, update the next oldest PR with the merged files and resolve issues.

This will end up creating many PRs, but because there is so much code to move (over 100KLoC) it's risky to expect a few large changes not introduce subtle bugs. Following the modularization the methods that are really large can be refactored further per file and in a more focused manner, with a PR per file since the code should be small enough to allow for that.

**Comment by Ruin0x11** [comment]
Seems like most of the work has been done, so I'll close. Thanks for merging everything.
